clap_complete/src/dynamic/completer.rs

```diff
@@ -26,7 +26,31 @@ pub trait Completer {
     ) -> Result<(), std::io::Error>;
 }
 
-/// Complete the given command
+/// NOTE: borrow the idea from Parser.rs::ParseState to record the state during completion.s
+#[derive(Debug, PartialEq, Eq, Clone)]
+enum CompletionState {
```

Comment:
style: we generally put implementation details for logic after the logic (so move this after `complete`)